Add ClaimCreatedEvent and ClaimModifiedEvent

Narimm opened the pull request:
1. Claim creation event... this is fired once all the claim creation checks have been passed.  It can be cancelled. It contains the claim and the creating player
2. Claim modification event - this is fired as the claim is modified - it cannot be cancelled. It contains the claim and the modifier...which can be null.

Additional changes
- The datastores basic claim create event is overloaded to allow us to differentiate between real new claims and modification of claims.
- Cleaned up the datastore imports..

Use Case:
This could be used so other plugins can hook gp and perform an action if a claim is created or changed. Something as simple as logging claim creations/modifications and deletions.

Narimm committed:
Create 2 new events
1. Claim creation event... this is fired once all the claim creation checks have been passed.  It can be cancelled. It contains the claim and the creating player
2. Claim modification event - this is fired as the claim is modified - it cannot be cancelled. It contains the claim and the modifier...which can be null.

Use Case:
This could be used so other plugins can hook gp and perform an action if a claim is created or changed. Something as simple as logging claim creations/modifications and deletions.

RoboMWM commented on src/main/java/me/ryanhamshire/GriefPrevention/DataStore.java:
```diff
@@ -19,14 +19,10 @@
-import org.bukkit.OfflinePlayer;
-import org.bukkit.World;
+import me.ryanhamshire.GriefPrevention.events.ClaimModifyEvent;
+import org.bukkit.*;
```
Can you make your IDE not do this please (wildcard imports)? Thanks.

RoboMWM reviewed (dismissed):
Looks good, will pull after the 1.13 PR.

Narimm committed:
Create 2 new events
1. Claim creation event... this is fired once all the claim creation checks have been passed.  It can be cancelled. It contains the claim and the creating player
2. Claim modification event - this is fired as the claim is modified - it cannot be cancelled. It contains the claim and the modifier...which can be null.

Use Case:
This could be used so other plugins can hook gp and perform an action if a claim is created or changed. Something as simple as logging claim creations/modifications and deletions.

RoboMWM committed:
Merge branch 'master' into newEvents
